restapi/src/main/java/io/stargate/web/docsapi/service/DocsSchemaChecker.java

```diff
@@ -0,0 +1,66 @@
+package io.stargate.web.docsapi.service;
+
+import com.github.benmanes.caffeine.cache.Cache;
+import com.github.benmanes.caffeine.cache.Caffeine;
+import io.stargate.web.docsapi.dao.DocumentDB;
+import io.stargate.web.docsapi.exception.ErrorCode;
+import io.stargate.web.docsapi.exception.ErrorCodeRuntimeException;
+import java.util.Objects;
+
+/**
+ * A helper class for checking that particular keyspace/table combinations exist and are valid
+ * document API collections.
+ */
+public class DocsSchemaChecker {
```

Comment:
I think you missed to impl the listening to the schema/table changes to invalidate the cache, or?

Reply:
Just finished the cache invalidation part!